src/bin/lttng/commands/add_context.c

```diff
@@ -240,6 +242,8 @@ const struct ctx_opts {
 	{ "preemptible", CONTEXT_PREEMPTIBLE },
 	{ "need_reschedule", CONTEXT_NEED_RESCHEDULE },
 	{ "migratable", CONTEXT_MIGRATABLE },
+	{ "callstack-kernel", CONTEXT_CALLSTACK_KERNEL },
```

Comment:
Since those contexts are only available on x86/x64 (to my knowledge?), we can add check for those archs
```
#if defined(__x86_64) || defined(__i386)
	{ "callstack-kernel", CONTEXT_CALLSTACK_KERNEL },
	{ "callstack-user", CONTEXT_CALLSTACK_USER },
#endif

This way we don't list an unsupported context on other architectures.

Reply:
CONTEXT_CALLSTACK_KERNEL is available on most (all?) architectures.
Only the CONTEXT_CALLSTACK_USER is only available on x86(64bit and 32bit)
I made the change you suggested for CONTEXT_CALLSTACK_USER